test/e2e/network_segmentation_default_network_annotation.go

```diff
@@ -103,4 +103,92 @@ var _ = Describe("Network Segmentation: Default network multus annotation", func
 			mac: "02:A1:B2:C3:D4:E5",
 		}),
 	)
+
+	Context("ValidatingAdmissionPolicy protection", func() {
+		It("should prevent adding, modifying and removing the default-network annotation on existing pods", func() {
+			if !isPreConfiguredUdnAddressesEnabled() {
+				Skip("ENABLE_PRE_CONF_UDN_ADDR not configured")
+			}
```

Comment:
Programmatic skips could make test getting skipped when not expected silently.
Please consider replacing programmatic skip with actual failure and change the test execution code to skip this test in case the env doesnt support it.

Reply:
While I agree that skipping tests is not ideal, it's not feasible at this time, considering the number of tests we have and all the lanes. This is a common approach we have to take to balance the number of lanes with the supported matrix of configurations.